### PR TITLE
Add PSM

### PR DESCRIPTION
**Pull request by eswak**
Add a simple Peg Stability Module that allows minting of `CREDIT` from a given collateral. The number of `CREDIT` minted or redeemed depends on the `ProfitManager.creditMultiplier()`. The PSM is important to prevent upward depegs (which could create bad debt in auctions even though the collateral of loans are sound), and it is also a convenient way to connect savers to borrowers : savers can mint through the PSM and enter rebase, while borrowers can borrow and directly redeem in the PSM for a peg asset more liquid than `CREDIT`.

Must merge first (PR diff might be hard to review otherwise, just look at commit b553b575ded141f44aa990868936caf35678d89a) : 
- #10 
- #15

### CODE REVIEW

File: src/loan/SimplePSM.sol [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    /// @notice redeem `amountIn` CREDIT for `amountOut` underlying tokens and send to address `to`
+    /// @dev see getRedeemAmountOut() to pre-calculate amount out
+    function redeem(
+        address to,
+        uint256 amountIn
+    ) external returns (uint256 amountOut) {
+        amountOut = getRedeemAmountOut(amountIn);
+        CreditToken(credit).transferFrom(msg.sender, address(this), amountIn);
+        CreditToken(credit).burn(amountIn);
```
Comment (ElliotFriedman) [inline src/loan/SimplePSM.sol:97]: `burnFrom` would do the same thing in less calls and save a warm SSTORE + SLOAD + call.

File: test/unit/loan/SimplePSM.t.sol [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    // - redeem moves a number of token equal to getRedeemAmountOut() i/o
+    // - rlcm buffer is depleted & replenished
+    // - rounding errors are within 1 wei for a mint/redeem round-trip
+    function testMintRedeem(uint256 input) public {
+        uint256 mintIn = input % 1e15 + 1; // [1, 1_000_000_000e6]
+
+        // update creditMultiplier
+        // this ensures that some fuzz entries will result in uneven divisions
+        // and will create a min/redeem round-trip error of 1 wei.
+        credit.mint(address(this), 100e18);
```
Comment (ElliotFriedman) [inline test/unit/loan/SimplePSM.t.sol:142]: why mint if the next step is just going to burn on 147?

File: src/loan/LendingTerm.sol [hunk 3]
```diff
@@ -660,7 +680,7 @@ contract LendingTerm is CoreRef {
         if (interest != 0) {
             // forward profit portion to the ProfitManager, burn the rest
             CreditToken(creditToken).transfer(profitManager, interest);
-            CreditToken(creditToken).burn(borrowAmount); // == loan.borrowAmount
+            CreditToken(creditToken).burn(principal);
```
[the remark this answers is not in the document]
Comment (eswak) [reply src/loan/LendingTerm.sol]: I think that's not part of this PR... maybe in one of the dependencies PR ? #10 #15

File: test/unit/loan/SimplePSM.t.sol (hunk 2 above)
Comment (eswak) [reply test/unit/loan/SimplePSM.t.sol:142]: to update the `creditMultiplier` the totalSupply needs to be != 0, so I'm minting a bunch of tokens, updating the `creditMultiplier`, and then burning them

**Pull request merged**